cpp/mrc/include/mrc/utils/type_utils.hpp

```diff
@@ -171,6 +172,12 @@ struct DataType
     TypeId m_type_id;
 };
 
+template <typename T>
+std::string boost_type_name()
+{
+    return boost::typeindex::type_id<T>().pretty_name();
+}
+
```

Comment:
How is this different than `type_name`?

Reply:
The boost implementation is more robust in terms of parsing out the type name and error checking. The current `type_name` implementation is hand rolled from the early days of neo, when I wasn't aware that Boost could do it. I think it makes the module traces a little more readable.